src/figuresClasses.ts

```diff
@@ -2,10 +1,100 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: string;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be greater than zero.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} cannot form a triangle.`);
+    }
+
+    this.color = color;
+    this.sideA = a;
+    this.sideB = b;
+    this.sideC = c;
+  }
+
+  getArea(): number {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+
+    const area = Math.sqrt(
+      s * (s - this.sideA) * (s - this.sideB) * (s - this.sideC),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  color: string;
+
+  radius: number;
 
-export class Triangle implements Figure {}
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Circle radius must be greater than zero.');
+    }
 
-export class Circle implements Figure {}
+    this.color = color;
+
+    this.radius = radius;
```

Comment:
Similar to the Triangle class, here you can apply access modifiers in the constructor parameters to implicitly create and assign class properties for 'color' and 'radius'.